torch/csrc/autograd/python_variable.cpp

```diff
@@ -2375,4 +2380,62 @@ c10::IntArrayRef concrete_sizes_fn(
   return c10::IntArrayRef(start, len);
 }
 
+c10::SymIntArrayRef concrete_sym_sizes_fn(
+    const c10::impl::PyInterpreter*,
+    const c10::TensorImpl* self) {
+  pybind11::gil_scoped_acquire gil;
+  at::impl::MaybeSetTLSOnEntryGuard guard;
+
+  std::cerr << "Running concrete_sym_sizes_fn\n";
+  auto out = torchDispatchFromTensorImpl(
+      self,
+      "size",
+      py::module::import("torch")
+          .attr("ops")
+          .attr("aten")
+          .attr("sym_size")
+          .attr("default")
+          .ptr(),
+      "torch.ops.aten");
+
+  std::cerr << "After torchDispatchFromTensorImpl\n";
+  if (out == Py_None) {
+    std::cerr << "Calling default sym_sizes_default\n";
+    return self->sym_sizes_default();
+  }
+
+  std::cerr << "Writing to buffer\n";
+
+  py::object values = py::reinterpret_steal<py::object>(out.ptr());
+
+  c10::TensorImpl* ptr = const_cast<c10::TensorImpl*>(self);
+  c10::optional<PyObject*> mb_obj = ptr->check_pyobj(getPyInterpreter());
+  TORCH_CHECK(
+      mb_obj.has_value(), "Tensor subclass's PyInterpreter has no value");
+  PyObject* subclass = *mb_obj;
+  Py_INCREF(subclass);
+  py::object sub = py::reinterpret_steal<py::object>(subclass);
+
+  // We need to squeeze SymIntNodes and ints into `SymInts`
+  // since it's a format `sym_sizes()` are stored in
+  py::list symints;
+  for (auto it = values.begin(); it != values.end(); it++) {
```

Comment:
What is begin/end doing on a plain PyObject?
If you expect a tuple, this is missing an assert.

Reply:
It can also be a tuple, a list or any iterable object that returns `SymInt`. Frankly, I don't know how to check that `if isinstance(e, Iterable):`  is true in C++ (more specifically, the `Iterable` part) but I'd think `values.begin()` would do this check for me at the expense of a slightly more confusing error message. Since we are the ones who will be implementing `__torch_dispatch__`, I figured that would be okay.

Reply:
You will get 
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'some_type' object is not iterable
````
It's not the best.
Also I would argue that we want the size function to always return a tuple of ints. So maybe we should just check that?
To do that, you can add a `TORCH_CHECK(py::isinstance<py::tuple>(obj), "...")`.

Note that if you add a TORCH_CHECK, you also want to add the `HANDLE_TH_ERRORS` and `END_HANDLE_TH_ERRORS_PYBIND` macros